### ISSUE TITLE
Average frame time in results

### ISSUE THREAD

**Issue by tweakimp** [Cryptyc/Sc2LadderServer#95]
Please add the average frame time and/or more statistics to the `results.json`.
Other things to add might be:
 - average actions sent
 - average actions*time/frame
 - time for first frame

**Comment by Archiatrus** [comment]
I propose we change the format of the result json a bit. Something like this:

```json
{
    "Results": [
        {
            "Bot1": 
                {
                    "Name": "Cpp",
                    "AverageLoopTime": 2,
                    "Actions": 873658974326
                }
            "Bot2": 
                {
                    "Name": "ocraft",
                    "AverageLoopTime": 3,
                    "Actions": 26
                }
            "Winner": "Cpp",
            "Map": "test.SC2Map",
            "Result": "Player1Win",
            "GameTime": 2787,
            "TimeStamp": "06-01-2019 17-06-27"
        }
    ]
}
```

As far as I can see one can only get actions on a individual unit basis. This means if I send an action to 10 marines to attack something as a group, it will be 10 actions. So it won't be comparable with human APM.

I am not a fan of redundant information. So if in the json there is already game loops and number of actions, imo we don't need average actions sent.

Time for first frame is a good idea.

**Comment by tweakimp** [comment]
I would also add the race. For random add Random(Race).

**Comment by spacekitteh** [comment]
Stuff like "total minerals gathered" "build order" etc would be neat to see too

**Comment by Archiatrus** [comment]
@spacekitteh total mineral gathered is available to the bot through the score interface. https://github.com/Blizzard/s2client-proto/blob/9906df71d6909511907d8419b33acc1a3bd51ec0/s2clientprotocol/score.proto#L31

The build order could also be logged by the bot itself.

I think the LM should only log things that are not available to the bot. Otherwise the result gets cluttered with lots of information that nobody really needs, that are just 'neat to see'. Just imagine the result json after 20 or 100 games if the build order is always included for both bots.

**Comment by alkurbatov** [comment]
@Archiatrus I believe this idea is inspired by Battle-net, where you can see gathering rate/count and actual build order. I doubt many bot authors would care about such things as BO logging especially because it could be considered as private strategy and requires additional efforts to implement such logging.
Another con for this is that we don't provide bot logs to third person, only to the author. And as you've said above such logs could be huge enough with hundreds of logged games.

**Comment by spacekitteh** [comment]
True, I keep forgetting people might not want to expose everything
